### PR TITLE
Interactive getting started tutorial

### PR DESCRIPTION
**Pull request by le1nux**
# What does this PR do?

This PR adds an interactive jupyter notebook-based Tutorial on how to use Modalities. 
We also added some fixes for warning output etc. 

## General Changes
* metrics are now printed only with a fixed number of decimals in the console during training. They are still logged to Weights and Biases with the full precision. 
*  removed abundant print statements

## Breaking Changes
* .. 

## Checklist before submitting final PR
- [x] My PR is minimal and addresses one issue in isolation
- [x] I have merged the latest version of the target branch into this feature branch
- [x] I have reviewed my own code w.r.t. correct implementation, missing type hints, proper documentation, etc.
- [x] I have run a sample config for model training
- [x] I have checked that all tests run through (`python tests/tests.py`)
- [x] I have updated the internal changelog (`CHANGELOG_DEV.md`)

### CODE REVIEW

File: src/modalities/models/model_factory.py [hunk 1]
```diff
@@ -64,10 +62,10 @@ def get_fsdp_wrapped_model(
             'FSDPTransformerAutoWrapPolicyFactory` is hardcoded and should be passed in instead.
             Different auto wrap policies may be supported in the future.
         """
-        print(
-            f"Unsharded number of parameters on rank {dist.get_rank()}: "
-            f"{get_local_number_of_trainable_parameters(model)}"
-        )
+        # print(
```
Comment (mali-git) [inline src/modalities/models/model_factory.py:65]: It would be great to print the number of parameters at least on one rank.

File: src/modalities/models/model_factory.py [hunk 2]
```diff
@@ -82,10 +80,10 @@ def get_fsdp_wrapped_model(
             sync_module_states=sync_module_states,
             use_orig_params=True,
         )
-        print(
-            f"Sharded number of parameters on rank {dist.get_rank()}:"
-            f"{get_local_number_of_trainable_parameters(fsdp_model)}"
-        )
+        # print(
```
Comment (mali-git) [inline src/modalities/models/model_factory.py:83]: It would be great to print the number of parameters at least on one rank.

File: src/modalities/trainer.py [hunk 3]
```diff
@@ -260,9 +260,9 @@ def train(
                     metrics=metrics,
                     # TODO: hardcoded metric key
                     throughput_metrics={
-                        "train samples/s": synced_num_samples_per_second,
-                        "train mfu": mfu,
-                        "lr mean": torch.tensor(scheduler.get_last_lr()).mean(),
+                        "train samples/s": ResultItem(synced_num_samples_per_second, 1),
+                        "train mfu": ResultItem(mfu, 2),
+                        "lr mean": ResultItem(torch.tensor(scheduler.get_last_lr()).mean(), 6),
```
Comment (mali-git) [inline src/modalities/trainer.py:265]: Since the learning rate can be arbitrarily small, I would log it in full precision.

**Review by mali-git — CHANGES_REQUESTED** [review]

File: src/modalities/trainer.py (hunk 3 above)
Comment (le1nux) [reply src/modalities/trainer.py]: done
